rsfbclient-rust/src/wire.rs

```diff
@@ -586,18 +586,19 @@ pub fn parse_sql_response(
     let mut data = Vec::with_capacity(xsqlda.len());
 
     for (col_index, var) in xsqlda.iter().enumerate() {
+        // Remove nullable type indicator
+        let sqltype = var.sqltype as u32 & (!1);
+
         if version >= ProtocolVersion::V13 && read_null(resp, col_index)? {
             // There is no data in protocol 13 if null, so just continue
             data.push(ParsedColumn::Complete(Column::new(
                 var.alias_name.clone(),
+                sqltype,
                 SqlType::Null,
             )));
             continue;
         }
 
-        // Remove nullable type indicator
-        let sqltype = var.sqltype as u32 & (!1);
-
         match sqltype {
```

Comment:
You need to remove the nullable indicator from the type for the match bellow to work, or else it would need to match both the nullable and non nullable versions of each type.

You can use `var.sqltype` to get the raw value for the `Column`.

Reply:
I am not remove the code , just move the code.

Reply:
![图片](https://github.com/user-attachments/assets/44ce5895-3a77-4622-a878-f469bd2aebb6)

Reply:
????

Reply:
![图片](https://github.com/user-attachments/assets/6541eea6-c12e-43bf-a568-0a4b7542b9e4)